File: insurance-frontend/src/app/core/model/CitizenPlan.ts
export interface CitizenPlan {
  citizenId?: number; // Primary Key [cite: 28]
  citizenName: string;
  gender: string;
  planName: string;
  planStatus: string;
  planStartDate?: string;
  planEndDate?: string;
  benefitAmount?: number;
  denialReason?: string;
  terminatedDate?: string;
  terminationReason?: string;
}

File: insurance-report/src/main/java/com/astik/insurance_report/entity/CitizenPlan.java
package com.astik.insurance_report.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "CITIZEN_PLANS_INFO")
@Data
public class CitizenPlan {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer citizenId;
    private String citizenName;
    private String gender;
    private String planName;
    private String planStatus;
    private LocalDate planStartDate;
    private LocalDate planEndDate;
    private Double benefitAmount;
    private String denialReason;
    private LocalDate terminatedDate;
    private String terminationReason;
}
